Dshop/apps/products_catalogue/tests/test_api_products.py

```diff
@@ -11,25 +11,55 @@ def create_category():
 
 
 @pytest.fixture
-def create_product():
+def create_active_product():
     category = Category.objects.create(name='Test Category', is_active=True)
     return Product.objects.create(
         name="main product",
         category=category,
         price=11,
         short_description="short desc",
-        full_description="full_description"
+        full_description="full_description",
+        is_active=True
     )
 
 
-@pytest.mark.django_db
-def test_access_protected_resource(api_client, user_instance_token):
+@pytest.fixture
+def create_inactive_product():
+    category = Category.objects.create(name='Test Category 2', is_active=True)
+    return Product.objects.create(
+        name="main product inactive",
+        category=category,
+        price=150,
+        short_description="short desc inactive",
+        full_description="full_description inactive",
+        is_active=False
+    )
+
+
+@pytest.fixture
+def authenticated_api_client(api_client, user_instance_token):
     api_client.credentials(HTTP_AUTHORIZATION=f'Token {user_instance_token.key}')
+    return api_client
 
+
+@pytest.mark.django_db
+def test_access_protected_resource(authenticated_api_client, create_active_product, create_inactive_product):
     url = reverse('products-api-list')
-    response = api_client.get(url)
+    response = authenticated_api_client.get(url)
     assert response.status_code == 200
 
+    results = response.data.get('results', [])
+    assert len(results) == 1
+
+    for product_data in results:
```

Comment:
Why using loop if there is only one product?
```suggestion
    product_data = results[0]
```
+ fix indentation

Reply:
also, it checks that fields are there, but not fields values. Would be good to add values check too. 
What I would to is prob create 

```python
def assert_active_object(data):
# copy all fields values and compare to data[x] ie:
    fields_values = {  
     "name": "main product"
     # ... more fields
     }
     for key, value in data.items():
         assert fields_values[key] == value
```
And use that function whenever you need to compare active_object with response.

Reply:
What do you think of this approach:

`def assert_active_object(data):
    fields_values = {
        "name": "main product",
        "price": "11.00",
        "short_description": "short desc",
        "full_description": "full_description",
    }
    for key, value in data.items():
        if key in fields_values:
            assert fields_values[key] == value`

because I don't want to check for example id or category

Reply:
Sorry for late response. 
For id it would be good to check if field is there, and for category too. Would be enough.

Reply:
Thanks @jacoor , I added this to test